### ISSUE TITLE
Is it able to output a probability or confidence for the segmentation result at every pixel?

### ISSUE THREAD

**Issue by wen-monster** [facebookresearch/segment-anything-2#318]
Limited by VRam, my mulit-obj tracking job need to be split. As a result, some objects may overlap. Therefore, can the model export not only 0or1 but also an float evaluation standard for later decision.

**Comment by heyoeyo** [comment]
The raw output of the model (referred to as `out_mask_logits` in the video notebook) is already a sort of confidence score for each pixel, where large positive values represent higher confidence. To make it more like a probability, you can pass it through a sigmoid function, to map it between 0.0 and 1.0:
```python
# Video segmentation loop
for out_frame_idx, out_obj_ids, out_mask_logits in predictor.propagate_in_video(inference_state):
  per_pixel_mask_prob = torch.sigmoid(out_mask_logits)
```

Written this way, the binary masked pixels would come from checking pixels with a value > 0.5. That is:
```python
binary_mask = torch.sigmoid(out_mask_logits) > 0.5
```
